src/ripple/rpc/handlers/LedgerHandler.cpp

```diff
@@ -212,13 +213,105 @@ doLedgerGrpc(RPC::GRPCContext<org::xrpl::rpc::v1::GetLedgerRequest>& context)
                 obj->set_mod_type(org::xrpl::rpc::v1::RawLedgerObject::DELETED);
             else
                 obj->set_mod_type(org::xrpl::rpc::v1::RawLedgerObject::CREATED);
+            auto blob = inDesired ? inDesired->slice() : inBase->slice();
```

Comment:
Can we add unit tests that excercises this new functionality?

Reply:
Yeah I'll add some tests

Reply:
Nit: could be `const`